File: 20-using-rest-proxy/src/spring-boot-restproxy-consumer-avro/src/main/java/com/trivadis/kafkaws/springbootkafkaconsumeravro/KafkaEventConsumer.java
package com.trivadis.kafkaws.springbootkafkaconsumeravro;

import com.trivadis.kafkaws.avro.v1.Notification;
import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Component;

@Component
public class KafkaEventConsumer {
    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaEventConsumer.class);

    @KafkaListener(topics = "${topic.name}", groupId = "simple-consumer-group")
    public void receive(ConsumerRecord<Long, Notification> consumerRecord) {
        Notification value = consumerRecord.value();
        Long key = consumerRecord.key();
        LOGGER.info("received key = '{}' with payload='{}'", key, value);
    }
}

File: 06c-working-with-avro-and-kotlin-springboot/src/spring-boot-kotlin-kafka-consumer-avro/src/main/kotlin/com/trivadis/kafkaws/springbootkotlinkafkaconsumer/KafkaEventConsumer.kt
package com.trivadis.kafkaws.springbootkotlinkafkaconsumer

import com.trivadis.kafkaws.avro.v1.Notification
import org.apache.kafka.clients.consumer.ConsumerRecord
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Component

@Component
class KafkaEventConsumer {
    @KafkaListener(topics = ["\${topic.name}"], groupId = "simple-kotlin-avro-consumer")
    fun receive(consumerRecord: ConsumerRecord<Long, Notification>) {
        println("received key = ${consumerRecord.key()} with payload=${consumerRecord.value()}")
    }
}